be/src/exec/pipeline/exchange/sink_buffer.h

```diff
@@ -177,10 +178,6 @@ class SinkBuffer {
             return;
         }
 
-        // Waiting for rpc task to exit, it won't take too long
-        if (_rpc_task_exit_promise != nullptr) {
-            _rpc_task_exit_promise->get_future().get();
-        }
```

Comment:
Why this wait is removed?

Reply:
Not necessary here, we just need to block and wait for the end in the destructor of SinkBuffer